ctapipe/utils/download.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import os
+from pathlib import Path
+import logging
+from tqdm import tqdm
+from urllib.parse import urlparse
+
+
+log = logging.getLogger(__name__)
+
+
+def download_file(url, path, auth=None, chunk_size=10240, progress=False):
+    """
+    Download a file. Will write to ``path + '.part'`` while downloading
+    and rename after successful download to the final name.
+
+    Parameters
+    ----------
+    url: str or url
+        The URL to download
+    path: pathlib.Path or str
+        Where to store the downloaded data.
+    auth: None or tuple of (username, password) or a request.AuthBase instance.
+    chunk_size: int
+        Chunk size for writing the data file, 10 kB by default.
+    """
+    log.info(f"Downloading {url} to {path}")
+
+    r = requests.get(url, stream=True, auth=auth)
+    name = urlparse(url).path.split("/")[-1]
+
+    # make sure the request is successful
+    r.raise_for_status()
+
+    path = Path(path)
+    path.parent.mkdir(parents=True, exist_ok=True)
+
+    # open a .part file to avoid a creating a broken file at the intended location
+    part_file = path.with_suffix(path.suffix + ".part")
+
+    total = float(r.headers.get("Content-Length", float("inf")))
+    pbar = tqdm(
+        total=total,
+        disable=not progress,
+        unit="B",
+        unit_scale=True,
+        desc=f"Downloading {name}",
+    )
+
+    try:
+        with part_file.open("wb") as f:
+            for chunk in r.iter_content(chunk_size=chunk_size):
+                f.write(chunk)
+                pbar.update(len(chunk))
+    except Exception:
+        # cleanup part file if something goes wrong
+        if part_file.is_file():
+            part_file.unlink()
+        raise
+
+    # when successful, move to intended location
+    part_file.rename(path)
+
+
+def get_cache_path(path, cache_name="lstchain"):
+    # need to make it relative
+    path = str(path).lstrip("/")
+    path = Path(os.environ["HOME"]) / ".cache" / cache_name / path
```

Comment:
Question to the mac users: Is this also an approproaiate location on mac?

Reply:
this is likely forbidden on the grid, where you can only write to the scratch disk. I would also avoid putting things in HOME in general, since that directory is backed up (and often quota-restricted) for most people, and these files should not be.

Reply:
Why should we event download these things on the GRID?

Reply:
But I will add an override for the cache.

Reply:
We could re-use the same functionality in the future for fetching things like calibration files, perhaps.  I'm not sure we'd ever run these tests on the grid, but in general it's best to support running anywhere we expect to run ctapipe. We already have the idea to do long-term tests on the grid, for example (though we'd use probably DIRAC to fetch the files needed for each job in that case).

Reply:
I added the env variable, which should also be useful for e.g. testing on our own testing machine with data available (although, the CTAPIPE_SVC_PATH could also be used for that)